vscode/src/completions/artificial-delay.test.ts

```
@@ -1,163 +1,191 @@
 import { afterEach, beforeEach, describe, expect, it, vi } from 'vitest'
 
+import type { FeatureFlagProvider } from '@sourcegraph/cody-shared'
+import { FeatureFlag, featureFlagProvider } from '@sourcegraph/cody-shared'
 import { getArtificialDelay, lowPerformanceLanguageIds, resetArtificialDelay } from './artificial-delay'
-
 const featureFlags = {
     user: true,
 }
 
-describe('getArtificialDelay', () => {
+describe('await getArtificialDelay', () => {
+    const enabledFeatureFlags = new Set<FeatureFlag>()
+    let originalFeatureFlagProviderInstance: FeatureFlagProvider | null
+
     beforeEach(() => {
         vi.useFakeTimers()
+        originalFeatureFlagProviderInstance = featureFlagProvider.instance
+        enabledFeatureFlags.clear()
+        featureFlagProvider.instance = {
+            evaluateFeatureFlag: (flag: FeatureFlag) => Promise.resolve(enabledFeatureFlags.has(flag)),
+            refresh: () => {},
+            getFromCache: (flagName: FeatureFlag) => {
+                return false
```

Oooh this looks subtle or wrong. Is this used? If not, let's throw here to fail noisily. If so, why would we consult `enabledFeatureFlags` the first time, but not when we're checking the cache?

---

Yeah it was used elsewhere but not here so I removed it.

---

Full disclosure I was writing some other tests which entailed mocking the feature flags and those tests would hit [this part of the codebase](https://sourcegraph.com/-/editor?remote_url=https%3A%2F%2Fgithub.com%2Fsourcegraph%2Fcody.git&branch=artificial-latency-fix&file=vscode%2Fsrc%2Fcompletions%2Fcompletion-provider-config.ts&editor=VSCode&version=2.2.17&start_row=48&start_col=26&end_row=48&end_col=26)  where you see getFromCache method being called so I had to specifically mock out that thing(NOTE: It tries to get the feature flag value from cache and that behavior is slightly different from evaluateFeatureFlag which also calls [getFromCache](https://sourcegraph.com/-/editor?remote_url=https%3A%2F%2Fgithub.com%2Fsourcegraph%2Fcody.git&branch=artificial-latency-fix&file=lib%2Fshared%2Fsrc%2Fexperimentation%2FFeatureFlagProvider.ts&editor=VSCode&version=2.2.17&start_row=127&start_col=32&end_row=127&end_col=32) here). 

I kept that thing here because I thought it was necessary but turns out that its not so I got rid of that and started failing loudly.